### ISSUE TITLE
Why is ModelMatrix assign = [1, 1, ..., 1] in v0.6.1?

### ISSUE THREAD

**Issue by jbrea** [JuliaStats/StatsModels.jl#133]
Is this intended?

```julia
(tmp) pkg> st
    Status `/tmp/Project.toml`
  [a93c6f00] DataFrames v0.19.0
  [38e38edf] GLM v1.3.0

julia> using GLM, DataFrames

julia> df = DataFrame(x = rand(12),
                      y = categorical(rand(1:3, 12)),
                      z = categorical(rand(1:2, 12)));

julia> f = @formula(x ~ y * z);

julia> model = fit(LinearModel, f, df, contrasts = Dict(:y => EffectsCoding(),
                                                        :z => EffectsCoding()));

julia> model.mm.assign
6-element Array{Int64,1}:
 1
 1
 1
 1
 1
 1
```

This used to be different before
```julia
(tmp) pkg> st
    Status `/tmp/Project.toml`
  [a93c6f00] DataFrames v0.19.0
  [38e38edf] GLM v1.3.0
  [3eaba693] StatsModels v0.6.1+ #c725015 (https://github.com/JuliaStats/StatsModels.jl.git)

julia> using GLM, DataFrames

julia> df = DataFrame(x = rand(12),
                      y = categorical(rand(1:3, 12)),
                      z = categorical(rand(1:2, 12)));

julia> f = @formula(x ~ y * z);

julia> model = fit(LinearModel, f, df, contrasts = Dict(:y => EffectsCoding(),
                                                        :z => EffectsCoding()));

julia> model.mm.assign
6-element Array{Int64,1}:
 0
 1
 1
 2
 3
 3
```

**Comment by kleinschmidt** [comment]
Not intended, thanks for the report.

**Comment by jbrea** [comment]
Replacing [this line](https://github.com/JuliaStats/StatsModels.jl/blob/v0.6.1/src/modelframe.jl#L177) with
```julia
asgn = mapreduce((it)->(first(it)-1)*ones(width(last(it))), append!, enumerate(model.mf.f.rhs.terms), init=Int[])
```
would bring back the old behaviour.

**Comment by kleinschmidt** [comment]
I have a slightly more robust fix that I'll push in a second...you have to take
into account the fact that the RHS might not be a `MatrixTerm` in some cases.

How much does it matter to you that `asgn` is zero-indexed?  I think the
original idea was that because the intercept term wasn't explicitly represented
in `Terms`, you don't want to index it.  But now all the terms are there,
including the Intercept term.

**Comment by jbrea** [comment]
Awesome, thanks for looking immediately into this.
I don't care if it is zero or one-indexed.

**Comment by kleinschmidt** [comment]
No problem, you caught me procrastinating :)

**Comment by kleinschmidt** [comment]
will be fixed in v0.6.2: JuliaRegistries/General#2114
